scripts/qa_crowbarsetup.sh

```diff
@@ -462,13 +464,53 @@ function cluster_node_assignment()
     wait_for 40 15 "! cat *.chef-client.ret | grep -qv '^0$'" "all chef-clients to succeed" \
         "cat *.chef-client.ret ; complain 73 'Manually triggered chef-client run failed on at least one node.'"
 
-    # assign nodes to clusters
-    clusternodesnetwork=`printf  "%s\n" $L | head -n$nodenumbernetworkcluster`
-    clusternodesservices=`printf "%s\n" $L | tail -n$nodenumberservicescluster`
+    ### Examples for clusterconfig:
+    # clusterconfig="data+services+network=2"
+    # clusterconfig="services+data=2:network=3:::"
+    # clusterconfig="services=3:data=2:network=2:"
+
+    for cluster in ${clusterconfig//:/ } ; do
+        [ -z "$cluster" ] && continue
+        local group=${cluster%=*}
+        local number=${cluster#*=}
+
+        local clustername=${group%%+*}
+        local nodes=
+
+        # clusternodesdata can only be the drbd nodes
+        if [[ $group =~ data ]] ; then
+            nodes="$clusternodesdrbd"
+        fi
+
+        # fetch nodes if not yet defined
+        if [ -z "$nodes" ] ; then
+            nodes=`printf  "%s\n" $L | head -n$number`
+        fi
 
-    for n in $clusternodesnetwork $clusternodesservices ; do
-        L=`printf "%s\n" $L | grep -iv $n`
+        # remove the nodes from the list of available nodes
+        for onenode in $nodes ; do
+            L=`printf "%s\n" $L | grep -iv $onenode`
```

Comment:
> Well actually it would mean a straight-forward s/line_by_line/puts/ would help a tiny

We use the printf stuff only with pipes. So the replacement is not that easy. The replacements will most likely be `var.each{...}` or `var.select{...}` or `var.some_other_array_function{...}` blocks.